client-java/src/main/java/uk/gov/gchq/palisade/client/java/JavaClient.java

```diff
@@ -0,0 +1,233 @@
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+
+        log.debug("Job created for token: {}", token);
+
+        return new Result() {
+
+        };
+
+    }
+
+    /**
+     * Handles the {@code ResourceReadyEvent} by scheduling a download. This method
+     * will return immediately as the download will be queued.
+     *
+     * @param event The event to be handled
+     */
+    @EventListener
+    public void onScheduleDownload(ResourceReadyEvent event) {
+        var resource = event.getResource();
+        var jobContext = getJobContext(resource.getToken());
+        var receiver = jobContext.getJobConfig().getReceiverSupplier().get();
+        clientContext.get(DownloadManager.class).schedule(resource, receiver);
+    }
+
+    /**
+     * Handles a download failed event
+     *
+     * @param event The event to be handled
+     */
+    @EventListener
+    public void onFailedDownload(DownloadFailedEvent event) {
+        log.debug(EVENT_CAUGHT, event);
+        log.debug("Download failed", event.getThrowble());
+    }
+
+    /**
+     * Handles the {@code ResourcesExhaustedEvent} event.
+     *
+     * @param event to be handled
+     */
+    @EventListener
+    public void onNoMoreDownloads(ResourcesExhaustedEvent event) {
+        log.debug(EVENT_CAUGHT, event);
+    }
+
+    public JobContext getJobContext(String token) {
+        return jobs.get(token);
+    }
+
+    /**
+     * Returns the {@code ApplicationContext} that was used to create this client.
+     * The context is used to retrieve other services that the client relies on,
+     * e.g. the {@code StateManager}.
+     *
+     * @return the {@code ApplicationContext} that was used to create this client.
+     * @see "https://docs.micronaut.io/latest/api/io/micronaut/context/ApplicationContext.html"
+     */
+    ClientContext getClientContext() {
+        return this.clientContext;
+    }
+
+    private PalisadeRequest createRequest(JobConfig jobConfig) {
+
+        var userId = jobConfig.getUserId();
+        var purpose = jobConfig.getPurpose();
+        var className = jobConfig.getClassname();
+        var requestId = jobConfig.getRequestId();
+        var resourceId = jobConfig.getResourceId();
+        var properties = jobConfig.getProperties();
+
+        var req = PalisadeRequest.builder()
+                .resourceId(resourceId)
+                .userId(UserId.builder()
+                        .id(userId)
+                        .build())
+                .requestId(RequestId.builder()
+                        .id(requestId)
+                        .build())
+                .context(Context.builder()
+                        .className(className)
+                        .purpose(purpose)
+                        .contents(properties)
+                        .build())
+                .build();
+
+        log.debug("new palisade request crteated from job config: {}", req);
```

Comment:
```suggestion
        log.debug("new palisade request created from job config: {}", req);
```

Reply:
My typing is awful. I'll be checking for spelling everywhere.